route-rs-runtime/src/io/pcap.rs

```diff
@@ -0,0 +1,212 @@
+    RunDump,
+    RunRealtime(FromPcapRealtimeState),
+}
+
+#[allow(dead_code)] // src_file
+struct FromPcapProcessor {
+    src_file: String,
+    capture: Peekable<PcapCaptureWrap>,
+    replay_state: PcapReplayState,
+}
+
+impl FromPcapProcessor {
+    fn new(src_file: String, replay_mode: PcapReplayMode) -> Self {
+        let capture = Capture::from_file(src_file.clone()).unwrap();
+        let mut cap_iter = PcapCaptureWrap { capture }.peekable();
+        let replay_state = match replay_mode {
+            PcapReplayMode::Dump => PcapReplayState::RunDump,
+            PcapReplayMode::Realtime => {
+                let start = Instant::now();
+                let cap_start = if let Some(p) = cap_iter.peek() {
+                    p.timestamp
+                } else {
+                    Duration::from_secs(0)
+                };
+                PcapReplayState::RunRealtime(FromPcapRealtimeState { start, cap_start })
+            }
+        };
+        FromPcapProcessor {
+            src_file,
+            capture: cap_iter,
+            replay_state,
+        }
+    }
+}
+
+struct PcapPacketWrap {
+    timestamp: Duration,
+    data: Vec<u8>,
+}
+// Wrapper to pcap::Capture to create iterator
+// because pcap::Packet is basically just references to file data
+struct PcapCaptureWrap {
+    capture: Capture<Offline>,
+}
+impl Iterator for PcapCaptureWrap {
+    type Item = PcapPacketWrap;
+    fn next(&mut self) -> Option<Self::Item> {
+        if let Ok(p) = self.capture.next() {
+            let ts = p.header.ts;
+            //Why are time_t and suseconds_t defined as signed integers?
+            let timestamp =
+                Duration::from_micros(ts.tv_sec as u64) + Duration::from_secs(ts.tv_sec as u64);
+            let data: Vec<u8> = p.data.into();
+            Some(PcapPacketWrap { timestamp, data })
+        } else {
+            None
+        }
+    }
+}
+use tokio::prelude::task;
+
+impl Stream for FromPcapProcessor {
+    type Item = Vec<u8>;
+    type Error = ();
+
+    fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
+        match &self.replay_state {
+            PcapReplayState::RunDump => {
+                if let Some(p) = self.capture.next() {
+                    let output_packet: Vec<u8> = p.data;
+                    Ok(Async::Ready(Some(output_packet)))
+                } else {
+                    Ok(Async::Ready(None))
+                }
+            }
+            PcapReplayState::RunRealtime(state) => {
+                let peek_intime = if let Some(p) = self.capture.peek() {
+                    (state.start.elapsed() > (p.timestamp - state.cap_start))
+                } else {
+                    //End of Pcap File
+                    return Ok(Async::Ready(None));
+                };
+                if peek_intime {
+                    if let Some(p) = self.capture.next() {
+                        let output_packet: Vec<u8> = p.data;
+                        println!(
```

Comment:
Lets' put this behind a debug switch or something rather than always print this out. It will clutter the test screen when many of these are running in parallel.